File: 2150.找出数组中的所有孤独数字.py
#
# @lc app=leetcode.cn id=2150 lang=python3
#
# [2150] 找出数组中的所有孤独数字
#

# @lc code=start
class Solution:
    def findLonely(self, nums: List[int]) -> List[int]:
        res = []
        d = Counter(nums)
        for k, v in d.items():
            if v == 1 and k - 1 not in d and k + 1 not in d:
                res.append(k)
        return res
# @lc code=end

File: 2150.找出数组中的所有孤独数字.go
/*
 * @lc app=leetcode.cn id=2150 lang=golang
 *
 * [2150] 找出数组中的所有孤独数字
 */

// @lc code=start
func findLonely(nums []int) []int {
	res := []int{}
	d := make(map[int]int)
	for _, v := range nums {
		d[v]++
	}
	for k, v := range d {
		_, ok1 := d[k-1]
		_, ok2 := d[k+1]
		if !ok1 && !ok2 && v == 1 {
			res = append(res, k)
		}
	}
	return res

}

// @lc code=end

